scenes/enumerated/problem_12724/problem_12724.py

```diff
@@ -0,0 +1,191 @@
+        task[-1].set_opacity(0)
+        task.to_edge(UP)
+        srr_rect = SurroundingRectangle(task, color=GREEN, buff=.2)
+        self.play(
+            Create(srr_rect),
+            Write(task)
+        )
+        self.wait()
+
+        num1 = MathTex("427", font_size=FONT_SIZE)
+        multiplication_sign = MathTex("\\times", font_size=FONT_SIZE-20)
+        num2 = MathTex("32", font_size=FONT_SIZE)
+
+        num2.next_to(num1, DOWN, aligned_edge=RIGHT)
+        multiplication_sign.next_to(VGroup(num1, num2), LEFT)
+        VGroup(num1, num2, multiplication_sign).next_to(srr_rect, DOWN, buff=.4, aligned_edge=LEFT).shift(RIGHT)
+        
+        line1 = Line().set_length(VGroup(num1, num2).width+.4)
+        line1.next_to(VGroup(num1, num2), DOWN)
+
+        self.play(Write(num1))
+        self.play(Write(num2))
+        self.play(Write(multiplication_sign))
+        self.play(GrowFromEdge(line1, LEFT))
+        self.wait()
+
+        first_number = MathTex(*"854", font_size=FONT_SIZE)
+        second_number = MathTex(*"1281", font_size=FONT_SIZE)
+
+        first_number.next_to(line1, DOWN).align_to(num2, RIGHT)
+        second_number.next_to(first_number, DOWN).align_to(first_number[-2], RIGHT)
+
+        avelacav1 = MathTex("1", font_size=FONT_SIZE, color=ORANGE).next_to(num1, LEFT, buff=2)
+        avelacav2 = MathTex("2", font_size=FONT_SIZE, color=ORANGE).next_to(num1, LEFT, buff=2)
+
+        for index in range(len(num1[0])):
+            digit_index = len(num1[0])-index-1
+            self.play(
+                Indicate(num1[0][digit_index]),
+                Indicate(num2[0][-1])
+            )
+            self.wait()
+            
+            if index == 0:
+                tex = MathTex(*"14", font_size=FONT_SIZE)
+                tex.next_to(line1, LEFT).shift(LEFT*5).align_to(first_number, DOWN)
+                self.play(tex.animate.shift(RIGHT*5))
+                self.wait()
+                self.play(
+                    ReplacementTransform(tex[0], avelacav1),
+                    ReplacementTransform(tex[1], first_number[len(first_number)-index-1])
+                )
+                self.wait()
+
+            if index == 1:
+                tex = MathTex("4", font_size=FONT_SIZE)
+                tex.next_to(line1, LEFT).shift(LEFT*5).align_to(first_number, DOWN)
+                self.play(tex.animate.align_to(avelacav1, RIGHT))
+                self.wait()
+                new_tex = MathTex("5", font_size=FONT_SIZE).move_to(tex.get_center())
+                self.play(ReplacementTransform(VGroup(tex, avelacav1), new_tex))
+                self.wait()
+                self.play(ReplacementTransform(new_tex, first_number[len(first_number)-index-1]))
+                self.wait()
+            
+            elif index == 2:
+                self.play(Write(first_number[len(first_number)-index-1]))
+                self.wait()
+
+        for index in range(len(num1[0])):
```

Comment:
էս մասի ֆոռ էղնելը կօգնե՞ հեչ

Reply:
for-ով էի գրել, որովհետև սկզբում վափշե սենց չէր, բոլորի համար գրեթե նույն բանն էր անում, բայց հետո լրիվ փոխվավ, ու հիմա միակ օգուտը էնա, որ մինչև if-ը գրած հատվածը անումա բոլորի համար